File: sdk/dotnet/lmsservice_db/src/api/certificates/CertificatesModel.cs

using System.ComponentModel.DataAnnotations;

namespace lmsservice_db.src.api.certificates.certificatesmodel;
public class CertificatesCreateModel
{
        [Required]
    [StringLength(64, MinimumLength = 0)]
    public string UserId { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string CertificateNumber { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string CertificateName { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string CertificateType { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string IssuedDate { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public DateTime ExpiryDate { get; set; } = DateTime.MinValue;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string CertificateUrl { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public float FinalGrade { get; set; } = 0f;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int CreditHours { get; set; } = 0;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Skills { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public bool IsVerified { get; set; } = false;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string VerificationUrl { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string IssuedBy { get; set; } = string.Empty;


}

public class CertificatesUpdateModel
{
        [Required]
    [StringLength(64, MinimumLength = 0)]
    public string UserId { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string CertificateNumber { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string CertificateName { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string CertificateType { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string IssuedDate { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public DateTime ExpiryDate { get; set; } = DateTime.MinValue;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string CertificateUrl { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public float FinalGrade { get; set; } = 0f;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int CreditHours { get; set; } = 0;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Skills { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public bool IsVerified { get; set; } = false;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string VerificationUrl { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string IssuedBy { get; set; } = string.Empty;


}

public class CertificatesSearchFilters
{
        [StringLength(64, MinimumLength = 0)]
    public string UserId { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string CertificateNumber { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string CertificateName { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string CertificateType { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string IssuedDate { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public DateTime ExpiryDate { get; set; } = DateTime.MinValue;

    [StringLength(64, MinimumLength = 0)]
    public string CertificateUrl { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public float FinalGrade { get; set; } = 0f;

    [StringLength(64, MinimumLength = 0)]
    public int CreditHours { get; set; } = 0;

    [StringLength(64, MinimumLength = 0)]
    public string Skills { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public bool IsVerified { get; set; } = false;

    [StringLength(64, MinimumLength = 0)]
    public string VerificationUrl { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string IssuedBy { get; set; } = string.Empty;


}

public class ApiResponse
{
    [Required]
    public Dictionary<string, object>? Data { get; set; }

    [StringLength(500, MinimumLength = 2)]
    public string Message { get; set; } = string.Empty;

    public bool? Success { get; set; }
}

File: sdk/python/src/lmsservice_db/api/certificates/certificates_model.py

from dataclasses import dataclass, field
from typing import Optional, Dict

@dataclass
class CertificatesCreateModel:
    UserId: str = field(metadata={"min_length": 0, "max_length": 64})
    CertificateNumber: str = field(metadata={"min_length": 0, "max_length": 64})
    CertificateName: str = field(metadata={"min_length": 0, "max_length": 64})
    CertificateType: str = field(metadata={"min_length": 0, "max_length": 64})
    IssuedDate: str = field(metadata={"min_length": 0, "max_length": 64})
    ExpiryDate: str = field(metadata={"min_length": 0, "max_length": 64})
    CertificateUrl: str = field(metadata={"min_length": 0, "max_length": 64})
    FinalGrade: str = field(metadata={"min_length": 0, "max_length": 64})
    CreditHours: str = field(metadata={"min_length": 0, "max_length": 64})
    Skills: str = field(metadata={"min_length": 0, "max_length": 64})
    IsVerified: str = field(metadata={"min_length": 0, "max_length": 64})
    VerificationUrl: str = field(metadata={"min_length": 0, "max_length": 64})
    IssuedBy: str = field(metadata={"min_length": 0, "max_length": 64})


@dataclass
class CertificatesUpdateModel:
    UserId: str = field(metadata={"min_length": 0, "max_length": 64})
    CertificateNumber: str = field(metadata={"min_length": 0, "max_length": 64})
    CertificateName: str = field(metadata={"min_length": 0, "max_length": 64})
    CertificateType: str = field(metadata={"min_length": 0, "max_length": 64})
    IssuedDate: str = field(metadata={"min_length": 0, "max_length": 64})
    ExpiryDate: str = field(metadata={"min_length": 0, "max_length": 64})
    CertificateUrl: str = field(metadata={"min_length": 0, "max_length": 64})
    FinalGrade: str = field(metadata={"min_length": 0, "max_length": 64})
    CreditHours: str = field(metadata={"min_length": 0, "max_length": 64})
    Skills: str = field(metadata={"min_length": 0, "max_length": 64})
    IsVerified: str = field(metadata={"min_length": 0, "max_length": 64})
    VerificationUrl: str = field(metadata={"min_length": 0, "max_length": 64})
    IssuedBy: str = field(metadata={"min_length": 0, "max_length": 64})


@dataclass
class CertificatesSearchParams:
    UserId: str = field(metadata={"min_length": 0, "max_length": 64})
    CertificateNumber: str = field(metadata={"min_length": 0, "max_length": 64})
    CertificateName: str = field(metadata={"min_length": 0, "max_length": 64})
    CertificateType: str = field(metadata={"min_length": 0, "max_length": 64})
    IssuedDate: str = field(metadata={"min_length": 0, "max_length": 64})
    ExpiryDate: str = field(metadata={"min_length": 0, "max_length": 64})
    CertificateUrl: str = field(metadata={"min_length": 0, "max_length": 64})
    FinalGrade: str = field(metadata={"min_length": 0, "max_length": 64})
    CreditHours: str = field(metadata={"min_length": 0, "max_length": 64})
    Skills: str = field(metadata={"min_length": 0, "max_length": 64})
    IsVerified: str = field(metadata={"min_length": 0, "max_length": 64})
    VerificationUrl: str = field(metadata={"min_length": 0, "max_length": 64})
    IssuedBy: str = field(metadata={"min_length": 0, "max_length": 64})


@dataclass
class ApiResponse:
    Data: Dict
    Message: Optional[str] = field(default=None, metadata={"min_length": 2, "max_length": 500})
    Success: Optional[bool] = None
